File: modules/pillo-data-privacy/ios/PilloDataPrivacyModule.swift
import ExpoModulesCore
import Foundation

public final class PilloDataPrivacyModule: Module {
  public func definition() -> ModuleDefinition {
    Name("PilloDataPrivacy")

    AsyncFunction("excludeSQLiteDatabaseFromBackupAsync") { (databasePath: String) in
      try [databasePath, "\(databasePath)-wal", "\(databasePath)-shm"].forEach(excludeFromBackup)
    }
  }

  private func excludeFromBackup(path: String) throws {
    var url = URL(fileURLWithPath: path)
    guard FileManager.default.fileExists(atPath: url.path) else { return }
    var values = URLResourceValues()
    values.isExcludedFromBackup = true
    try url.setResourceValues(values)
  }
}

File: modules/pillo-data-privacy/src/PilloDataPrivacyModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class PilloDataPrivacyModule extends NativeModule<{}> {
  excludePathFromBackupAsync(path: string): Promise<void>;
  excludeSQLiteDatabaseFromBackupAsync(databasePath: string): Promise<void>;
}

export default requireNativeModule<PilloDataPrivacyModule>('PilloDataPrivacy');
